src/sha512.cpp

```diff
@@ -12,13 +17,17 @@
  * Tom St Denis, tomstdenis@gmail.com, http://libtom.org
  */
 
-#include "fixedint.h"
-#include "sha512.h"
+typedef std::uint64_t u64;
+typedef std::int64_t i64;
+typedef std::int32_t i32;
```

Comment:
would you mind using the ``using`` keyword instead? I think it's  more intuitive. i.e. ``using u64 = std::uint64_t;``